pkg/cli/subcommands/check/command.go

```diff
@@ -217,18 +207,14 @@ func MakeCommand(globalParamsGetter func() GlobalParams) *cobra.Command {
 				// TODO(components): this is a temporary hack as the StartServer() method of the API package was previously called with nil arguments
 				// This highlights the fact that the API Server created by JMX (through ExecJmx... function) should be different from the ones created
 				// in others commands such as run.
-				fx.Provide(func() flare.Component { return nil }),
+				fx.Supply(optional.NewNoneOption[rcservice.Component]()),
+				fx.Supply(optional.NewNoneOption[rcservicemrf.Component]()),
+				fx.Supply(optional.NewNoneOption[logagent.Component]()),
```

Comment:
small question why we went from this one `fx.Provide(func() optional.Option[logagent.Component] {
					return optional.NewNoneOption[logagent.Component]()

				}),` to this one: `fx.Supply(optional.NewNoneOption[logagent.Component]()),`

Reply:
It is shorter and it does the same thing 😄